model-api-gen/src/test/kotlin/org/modelix/metamodel/generator/KotlinGeneratorTest.kt

```diff
@@ -81,6 +82,50 @@ class KotlinGeneratorTest {
         // val outputDir = File(".").toPath().resolve("build").resolve("test-generator-output")
         val outputDir = File("build/test-generator-output").toPath()
         TypescriptMMGenerator(outputDir).generate(LanguageSet(listOf(language)).process())
+
+        val indexFileContents = outputDir.resolve("index.ts").readText()
+        assertFalse(indexFileContents.contains("export * from"), "Does not include barrels when not explicitly opted in.")
+    }
+
+    @Test
+    fun test_ts_with_barrels() {
+        val input = """
+            name: org.modelix.entities
+            concepts:
+            - name: Entity
+              properties:
+              - name: name
+              children:
+              - name: properties
+                type: Property
+                multiple: true
+                optional: true
+            - name: Property
+              children:
+              - name: type
+                type: Type
+                optional: false
+            - name: Type
+            - name: EntityType
+              extends:
+              - Type
+              references:
+              - name: entity
+                type: Entity
+                optional: false
+            enums: []
+        """.trimIndent()
+
+        val language = Yaml.default.decodeFromString<LanguageData>(input)
+        val outputDir = File("build/test-generator-output").toPath()
```

Comment:
We should use a virtual file system here or even seperate creating files from writing them to disk. Out of scope for your feature, because we do it in all other test also.